stock_picking_batch_creation/wizards/make_picking_batch.py

```diff
@@ -205,11 +222,21 @@ def gt(value1, value2, digits):
             picking_device = self._compute_device_to_use(picking)
             if device != picking_device:
                 continue
-
             picking._init_nbr_bins_on_device_field(device)
-            available_bins_outreached = (
-                available_nbr_bins - picking.nbr_bins_batch_picking < 0
-            )
+            if self.group_pickings_by_partner:
+                (
+                    bins_for_pickings_by_partner,
+                    volume_and_bins_by_partners,
+                ) = self._get_nbr_bins_by_partner(
+                    picking, device, volume_and_bins_by_partners
+                )
+                available_bins_outreached = (
+                    available_nbr_bins - bins_for_pickings_by_partner < 0
+                )
+            else:
+                available_bins_outreached = (
+                    available_nbr_bins - picking.nbr_bins_batch_picking < 0
+                )
```

Comment:
seems too complicated and could be simplified...
```python 
    nbr_bins = picking.nbr_bins_batch_picking
    old_volume = volume_by_partner[picking.partner_id]
    new_volume = volume_by_partner[picking.partner_id] + picking.total_volume_batch_picking
    if self.group_pickings_by_partner:
        nbr_bins = math.ceil(new_volume / device.volume_per_bin) - math.ceil(old_volume / device.volume_per_bin)
    
    available_bins_outreached = (
        available_nbr_bins - nbr_bins < 0
    )
    if available_bins_outreached:
        continue
    ...
    ...
    available_nbr_bins -= nbr_bins
    volume_by_partner[picking.partner_id]  = new_volume

...
used_nbr_bins = device.nbr_bins - available_nbr_bins
return selected_pickings, unselected_picking, used_nbr_bins
```

moreover the volume was summed into the total by partner even if the picking was excluded due to some limits reached... You should add a test for this case....